### PR TITLE
Update Data Prepper copy-values processor docs

### LINKED ISSUE #6390: [DOC] Update docs for copy_value processor in Data Prepper
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 New options being added to the processor: https://github.com/opensearch-project/data-prepper/pull/4085

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by oeyh**
### Description
Adds a few new options to copy-values processor and rewrite the examples.

### Issues Resolved
Closes #6390 

### Version
Since Data Prepper 2.7

### Frontend features
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 1]
```diff
@@ -14,44 +14,125 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+```json
+{
+  "mylist": [
+    {"name": "apple", "color": "red"},
+    {"name": "orange", "color": "orange"}
+  ]
+}
+```
+
+The processed event will contain "newlist" with selectively copied fields:
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md]:
```suggestion
The processed event contains `newlist` with selectively copied fields:
```

**Review by vagimeli — APPROVED** [review]
Doc review completed. Minimal edits. I scheduled it for editorial. Upon that approval, I'll merge the PR. Thank you.

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 2]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+{
+  "mylist": [
+    {"name": "apple", "color": "red"},
+    {"name": "orange", "color": "orange"}
+  ]
+}
 ```
 
-If `newMessage` already exists, its existing value is overwritten with `value`.
+The processed event contains "newlist" with selectively copied fields:
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:126]:
```suggestion
The processed event contains `newlist` with selectively copied fields:
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 3]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
 
-| Option | Required | Description |
-:--- | :--- | :---
-| `entries` | Yes | A list of entries to be copied in an event. |
-| `from_key` | Yes | The key of the entry to be copied. |
-| `to_key` | Yes | The key of the new entry to be added. |
-| `overwrite_if_to_key_exists` | No | When set to `true`, the existing value is overwritten if `key` already exists in the event. The default value is `false`. |
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `entries` | Yes | [entry](#entry) | A list of entries to be copied in an event. See [entry](#entry) for details. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:19]:
```suggestion
| `entries` | Yes | [entry](#entry) | A list of entries to be copied in an event. See [entry](#entry) for more information. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 4]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `from_list` | No | String | The key of the list of objects to be copied. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:20]: The key "for"?

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 5]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `to_list` | No | String | The key of the new list to be added. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:21]: The key "for"?

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 6]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `overwrite_if_to_list_exists` | No | Boolean | When set to `true`, the existing value is overwritten if the `key` specified by `to_list` already exists in the event. Default is `false`. |
+
+## entry
+
+For each entry, you can configure the following options.
+
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `from_key` | Yes | String | The key of the entry to be copied. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:30]: The key "for"?

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 7]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `to_key` | Yes | String | The key of the new entry to be added. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:31]: The key "for"?

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 8]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `overwrite_if_to_key_exists` | No | Boolean | When set to `true`, the existing value is overwritten if `key` already exists in the event. Default is `false`. |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:32]:
```suggestion
| `overwrite_if_to_key_exists` | No | Boolean | When set to `true`, the existing value is overwritten if the `key` already exists in the event. Default is `false`. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 9]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+
 
 ## Usage
 
-To get started, create the following `pipeline.yaml` file:
+The following examples show how the `copy_values` processor can be used in different cases.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:37]:
```suggestion
The following examples show you how to use the `copy_values` processor.
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 10]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+
+### Example: Copy values and skip existing fields
+
+The following example shows you how to configure the processor to copy values and not overwrite existing fields:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:41]: Should "and not overwrite" be "and skip" to match the heading?

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 11]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+{
+  "mylist": [
+    {"name": "apple", "color": "red"},
+    {"name": "orange", "color": "orange"}
+  ]
+}
 ```
 
-If `newMessage` already exists, its existing value is overwritten with `value`.
+The processed event contains `newlist` with selectively copied fields:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:126]:
```suggestion
The processed event contains a `newlist` with selectively copied fields:
```

**Review by natebower — COMMENTED** [review]
@oeyh @vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

**Comment by oeyh** [comment]
@natebower @vagimeli The changes look good to me. Thank you for the review!

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 12]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
-| Option | Required | Description |
-:--- | :--- | :---
-| `entries` | Yes | A list of entries to be copied in an event. |
-| `from_key` | Yes | The key of the entry to be copied. |
-| `to_key` | Yes | The key of the new entry to be added. |
-| `overwrite_if_to_key_exists` | No | When set to `true`, the existing value is overwritten if `key` already exists in the event. The default value is `false`. |
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `entries` | Yes | [entry](#entry) | A list of entries to be copied in an event. See [entry](#entry) for more information. |
+| `from_list` | No | String | The key of the list of objects to be copied. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:20]:
```suggestion
| `from_list` | No | String | The key for the list of objects to be copied. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 13]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `to_list` | No | String | The key of the new list to be added. |
+| `overwrite_if_to_list_exists` | No | Boolean | When set to `true`, the existing value is overwritten if the `key` specified by `to_list` already exists in the event. Default is `false`. |
+
+## entry
+
+For each entry, you can configure the following options.
+
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `from_key` | Yes | String | The key of the entry to be copied. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:30]:
```suggestion
| `from_key` | Yes | String | The key for the entry to be copied. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 14]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
-:--- | :--- | :---
-| `entries` | Yes | A list of entries to be copied in an event. |
-| `from_key` | Yes | The key of the entry to be copied. |
-| `to_key` | Yes | The key of the new entry to be added. |
-| `overwrite_if_to_key_exists` | No | When set to `true`, the existing value is overwritten if `key` already exists in the event. The default value is `false`. |
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `entries` | Yes | [entry](#entry) | A list of entries to be copied in an event. See [entry](#entry) for more information. |
+| `from_list` | No | String | The key for the list of objects to be copied. |
+| `to_list` | No | String | The key of the new list to be added. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:21]:
```suggestion
| `to_list` | No | String | The key for the new list to be added. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 15]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+| `overwrite_if_to_list_exists` | No | Boolean | When set to `true`, the existing value is overwritten if the `key` specified by `to_list` already exists in the event. Default is `false`. |
+
+## entry
+
+For each entry, you can configure the following options.
+
+| Option | Required | Type | Description |
+:--- | :--- | :--- | :---
+| `from_key` | Yes | String | The key for the entry to be copied. |
+| `to_key` | Yes | String | The key of the new entry to be added. |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:31]:
```suggestion
| `to_key` | Yes | String | The key for the new entry to be added. |
```

File: _data-prepper/pipelines/configuration/processors/copy-values.md [hunk 16]
```diff
@@ -14,44 +14,126 @@ The `copy_values` processor copies values within an event and is a [mutate event
...
+
 
 ## Usage
 
-To get started, create the following `pipeline.yaml` file:
+The following examples show you how to use the `copy_values` processor.
+
+### Example: Copy values and skip existing fields
+
+The following example shows you how to configure the processor to copy values and not overwrite existing fields:
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/copy-values.md:41]:
```suggestion
The following example shows you how to configure the processor to copy values and skip existing fields:
```

**Comment by vagimeli** [comment]
> @natebower @vagimeli The changes look good to me. Thank you for the review!

Editorial feedback addressed and changes completed.

**Pull request merged**